src/main/java/br/com/caelum/vraptor/test/requestflow/UserFlow.java

```diff
@@ -114,11 +109,26 @@ public UserFlow to(final String url, final HttpMethod httpMethod, final Paramete
 			@Override
 			public VRaptorTestResult call(HttpSession session) {
 				LOG.debug("starting request to " + url);
-				request = new MockHttpServletRequest(context, httpMethod.toString(), url) {
+				final String localUrl;
+				final String query;
+				if(url.contains("?")){
+					String[] parts = url.split("\\?");
+					localUrl = parts[0];
+					query = parts[1];
+				} else {
+					localUrl = url;
+					query = "";
+				}
```

Comment:
i'd just extract some methods to make it clearer. i.e
```
String query = extractQueryFrom(url);
```
and the possible impl would look like this:
```
protected String extractQueryFrom(String url) {
    if (!url.contains("?")) {
        return "";
    }
    return url.split("\\?")[1];
}
```
what do you think @clairton?  besides that everything looks great, thanks!